core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -637,7 +637,8 @@ boolean isPreserveAliases() {
                     throw infe(expression);
                 }
 
-                if (expression.contains("*")) {
+                if (Regex.isSimpleMatchPattern(expression) &&
+                        (options.expandWildcardsOpen() || options.expandWildcardsClosed())) {
```

Comment:
I actually reverted the change to add the check, because if wildcards are both disabled then this check is already handled at a higher level, in `resolve` instead of `innerResolve`:

https://github.com/elastic/elasticsearch/blob/e57720e09170208e8059c39d43b848024d79ed9d/core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L559-L561